File: Source/Bifrost.JavaScript.Specs/utils/for_StringMapping/when_matching_a_string_with_only_a_wildcard.js
describe("when matching a string with only a wildcard", function () {
    var mapping = Bifrost.StringMapping.create({
        format: "**/",
        mappedFormat: "**."
    });
    var result = mapping.matches("this/is/a/wildcard/uri/for_things");

    it("should match", function () {
        expect(result).toBe(true);
    });
});

File: Source/Bifrost.Specs/Utils/for_StringMapping/when_matching_a_string_with_only_a_wildcard.cs
using Bifrost.Utils;
using Machine.Specifications;

namespace Bifrost.Specs.Utils.for_StringMapping
{
    public class when_matching_a_string_with_only_a_wildcard
    {
        static StringMapping mapping = new StringMapping(
                "**/",
                "**."
            );
        static bool result;

        Because of = () => result = mapping.Matches("this/is/a/wildcard/uri/for_things");

        It should_match = () => result.ShouldBeTrue();
    }
}
